### PR TITLE
release: v1.24.0-rc4

### PR DESCRIPTION
**Pull request by jennijuju**
backports
#11371
#11354 
#11373 
#11357 

this rc also sets the mainnet upgrade at epoch 3431940

### CODE REVIEW

File: build/params_mainnet.go [hunk 1]
```diff
@@ -95,8 +95,8 @@ const UpgradeLightningHeight = 2809800
 // 2023-05-18T13:00:00Z
 const UpgradeThunderHeight = UpgradeLightningHeight + 2880*21
 
-// ???????
-var UpgradeWatermelonHeight = abi.ChainEpoch(9999999999)
+// 2023-11-29T13:30:00Z
+var UpgradeWatermelonHeight = abi.ChainEpoch(3431940)
```
Comment (Stebalien) [inline build/params_mainnet.go:99]: AKA

> PST Wed Nov 29 05:30:00 2023
> UTC Wed Nov 29 13:30:00 2023

**Review by Stebalien — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-
-**This release candidate does NOT set an upgrade epoch for mainnet. However, it includes a patch for the calibration testnet to fix an issue where partitions with expired sectors had to be compacted before they could be moved. Unfortunately, this bug impacts consensus and necessitates a coordinated upgrade on the Calibration network to deploy the new code. The calibration network is scheduled to upgrade all miner actors to the new, fixed miner actor CID at epoch 1070494, which is expected to occur at 2023-11-07T13:00:00Z.**
+**It includes a patch for the calibration testnet to fix an issue where partitions with expired sectors had to be compacted before they could be moved. Unfortunately, this bug impacts consensus and necessitates a coordinated upgrade on the Calibration network to deploy the new code. The calibration network is scheduled to upgrade all miner actors to the new, fixed miner actor CID at epoch 1070494, which is expected to occur at 2023-11-07T13:00:00Z.**
 
+This release candidate also sets an upgrade epoch for mainnet at `3431940` 2023-11-29T13:30:00Z.
```
Comment (arajasek) [inline CHANGELOG.md:7]: I'd suggest moving this above the calibnet blurb, it's what most users will care about (I think?)

**Comment by arajasek** [comment]
Can I ask someone to examine the v12.tar.zst file with this PR, and confirm that:
- it includes TWO calibrationnet bundles, called `builtin-actors-calibrationnet.car` and `builtin-actors-calibrationnet-buggy.car`
- `builtin-actors-calibrationnet.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`
- `builtin-actors-calibrationnet-buggy.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`
- `builtin-actors-mainnet.car` has a manifest CID of `bafy2bzacecooh5cjxkmraawn7wvmrx7k5hwdmbvqdteujr33mnaeclsazyb6a`

**Review by arajasek — APPROVED** [review]

**Comment by jennijuju** [comment]
> * `builtin-actors-calibrationnet.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`
> * `builtin-actors-calibrationnet-buggy.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`

Im being paranoid as you demanded and check the bundles. Calib buggy and mainnet matches what you suggested - and I assume you meant a different calibration non buggy cid which ends `kzru`(in your message they are dupes)

**Comment by arajasek** [comment]
> > * `builtin-actors-calibrationnet.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`
> > * `builtin-actors-calibrationnet-buggy.car` has a manifest CID of `bafy2bzacedrunxfqta5skb7q7x32lnp4efz2oq7fn226ffm7fu5iqs62jkmvs`
> 
> Im being paranoid as you demanded and check the bundles. Calib buggy and mainnet matches what you suggested - and I assume you meant a different calibration non buggy cid which ends `kzru`(in your message they are dupes)

I did in fact mean that! Thank you! :blue_heart:

**Pull request merged**